src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/Solderer.java

```diff
@@ -0,0 +1,160 @@
+        @Example(".pattern(item('minecraft:clay')).input(item('minecraft:gold_ingot')).output(item('minecraft:diamond')).gold(5).time(100)"),
+        @Example(".pattern(item('minecraft:coal_block')).output(item('minecraft:nether_star')).gold(75)"),
+    })
+    public Solderer.RecipeBuilder recipeBuilder() {
+        return new Solderer.RecipeBuilder();
+    }
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(x -> SoldererManager.removeRecipe(x.getPattern(), x.getAdditive(), 9999));
+        restoreFromBackup().forEach(SoldererManager::addRecipe);
+    }
+
+    public void add(SoldererManager.SoldererRecipe recipe) {
+        if (recipe == null) return;
+        addScripted(recipe);
+        SoldererManager.addRecipe(recipe);
+    }
+
+    public boolean remove(SoldererManager.SoldererRecipe recipe) {
+        if (recipe == null) return false;
+        addBackup(recipe);
+        return SoldererManager.removeRecipe(recipe.getPattern(), recipe.getAdditive(), 9999) != null;
+    }
+
+    @MethodDescription(example = @Example("item('prodigytech:pattern_circuit_refined')"))
+    public boolean removeByPattern(IIngredient pattern) {
+        SoldererManager.SoldererRecipe recipe = SoldererManager.RECIPES.stream()
+                .filter(r -> pattern.test(r.getPattern()))
+                .findFirst().orElse(null);
+        return remove(recipe);
+    }
+
+    @MethodDescription(example = @Example("item('minecraft:iron_ingot')"))
+    public boolean removeByAdditive(IIngredient additive) {
+        SoldererManager.SoldererRecipe recipe = SoldererManager.RECIPES.stream()
+                .filter(r -> r.requiresAdditive() && additive.test(r.getAdditive()))
+                .findFirst().orElse(null);
+        return remove(recipe);
+    }
+
+    @MethodDescription(example = @Example("item('prodigytech:circuit_refined')"))
+    public boolean removeByOutput(IIngredient output) {
+        SoldererManager.SoldererRecipe recipe = SoldererManager.RECIPES.stream()
+                .filter(r -> output.test(r.getOutput()))
+                .findFirst().orElse(null);
+        return remove(recipe);
+    }
+
+    @MethodDescription(priority = 2000, example = @Example(commented = true))
+    public void removeAll() {
+        SoldererManager.RECIPES.forEach(this::addBackup);
+        SoldererManager.removeAll();
+    }
+
+    @MethodDescription(priority = 2000, example = @Example(commented = true))
+    public void removeWithoutAdditive() {
+        SoldererManager.RECIPES.removeIf(r -> {
+            if (r.requiresAdditive()) return false;
+            addBackup(r);
+            return true;
+        });
+    }
+
+    @MethodDescription(type = MethodDescription.Type.QUERY)
+    public SimpleObjectStream<SoldererManager.SoldererRecipe> streamRecipes() {
+        return new SimpleObjectStream<>(SoldererManager.RECIPES)
+                .setRemover(this::remove);
+    }
+
+    @Property(property = "input", valid = @Comp("1"))

```

Comment:
because the actual allowed value is a range, this should be more like
```suggestion
    @Property(property = "input", valid = {@Comp(type = Comp.Type.GTE, value = "0"), @Comp(type = Comp.Type.LTE, value = "1")})
```